src/backend/main.ts

```diff
@@ -1168,6 +1176,75 @@ ipcMain.handle(
   }
 )
 
+async function runBeforeLaunchScript(
+  gameInfo: GameInfo,
+  gameSettings: GameSettings
+) {
+  if (gameSettings.beforeLaunchScriptPath) {
+    return new Promise((resolve, reject) => {
+      logInfo(
+        [
+          'Running script before',
+          gameInfo.title,
+          `(${gameSettings.beforeLaunchScriptPath})`
+        ],
+        LogPrefix.Backend
+      )
+
+      // Execute script before launch, wait until the script
+      // exits to continue
+      //
+      // The script can start sub-processes with `bash another-command &`
+      // if `another-command` can run async
+      const child = spawn(gameSettings.beforeLaunchScriptPath, {
+        cwd: gameInfo.install.install_path,
+        ...execOptions
+      })
+
+      child.stdout.on('data', (data) => {
+        logInfo(data.toString(), LogPrefix.Backend)
+      })
+
+      child.stderr.on('data', (data) => {
+        logInfo(data.toString(), LogPrefix.Backend)
+      })
+
+      child.on('exit', () => {
+        resolve(true)
+      })
+
+      child.on('error', (err: Error) => {
+        logError(
+          ['Error running before script: ', err.message],
+          LogPrefix.Backend
+        )
+        if (err.stack) logError(err.stack, LogPrefix.Backend)
+        reject(err.message)
+      })
+    })
+  } else {
+    return Promise.resolve(true)
+  }
+}
+
+function runAfterLaunchScript(gameInfo: GameInfo, gameSettings: GameSettings) {
+  if (gameSettings.afterLaunchScriptPath) {
+    logInfo(
+      [
+        'Running script after',
+        gameInfo.title,
+        `(${gameSettings.afterLaunchScriptPath})`
+      ],
+      LogPrefix.Backend
+    )
+    const output = execSync(gameSettings.afterLaunchScriptPath, {
```

Comment:
Using `execSync` here isn't *that* great since it blocks the Backend thread. Ideally use `exec` with a callback like
```ts
(error, stdout) => {
  if (error) logError(error, LogPrefix.Backend)
  logInfo(stdout, LogPrefix.Backend)
}